src/crimson/os/seastore/onode.h

```diff
@@ -14,8 +14,14 @@
 namespace crimson::os::seastore {
 
 struct onode_layout_t {
+  // around 350 bytes for fixed fields in object_info_t,
+  // the left are for the variable-sized fields like oid
```

Comment:
I don't think we need extra space in `onode_layout_t` to store oid, as it is already present in the tree as the key. For example, we can get the oid from Cursor::get_ghobj(), see https://github.com/ceph/ceph/blob/master/src/crimson/os/seastore/onode_manager/staged-fltree/tree.h#L80-L84

Reply:
That's out of scope here.  The issue is that the OSD's own object_info_t structure (which it stores in the OI_ATTR) encodes the full hobject_t.  A good next project would be to rework the OSD's usage of this interface to remove that requirement and thus shrink the object_info_t.

Reply:
Hmm, looks `object_info_t` is encoded outside `ObjectStore`, need to adjust OSD logic to exclude oid from the OI buffer.

Reply:
Right.

Reply:
@xxhdx1985126 could add a FIXME here to remember that we want to exclude the oid in the future?

Reply:
Ok